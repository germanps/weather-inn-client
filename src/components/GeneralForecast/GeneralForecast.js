import React from 'react'
import { EuiFlexGroup, EuiFlexItem, EuiSpacer } from '@elastic/eui'
import PropTypes from 'prop-types'
import './GeneralForecast.scss'

export default function GeneralForecast({ weatherToday, weatherTomorrow }) {
    return (
        <>
            <EuiFlexGroup className="general-forecast">
                <EuiFlexItem className="general-forecast__column">
                    <EuiSpacer />
                    <h3 className="subtitle">Previsto para hoy</h3>
                    <p className="text">{weatherToday}</p>
                </EuiFlexItem>
                <EuiFlexItem className="general-forecast__column">
                    <EuiSpacer />
                    <h3 className="subtitle">Previsión para mañana</h3>
                    <p className="text">{weatherTomorrow}</p>
                </EuiFlexItem>

            </EuiFlexGroup>
        </>
    )
}

GeneralForecast.propTypes = {
    weatherToday: PropTypes.array,
    weatherTomorrow: PropTypes.array
}
