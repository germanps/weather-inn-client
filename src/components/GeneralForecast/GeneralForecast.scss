@import '../../scss/index.scss';

.general-forecast{

    &__column{
        
        .subtitle-low{
            @include subTitle($font-size * .7, $danger, left, .5rem, bold)

        }
        .text{
            font-size: $font-size * .65;
            line-height: 14px;
        }
    }
}