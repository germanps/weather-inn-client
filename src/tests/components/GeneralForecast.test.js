import React from 'react'
import { shallow } from 'enzyme'
import PropTypes from 'prop-types'
import GeneralForecast from '../../components/GeneralForecast'

describe('Test in <GeneralForecast />', () => {
    const weatherToday = ['today']
    const weatherTomorrow = ['tomorrow']
    test('should show the component correctly', () => {
        const wrapper = shallow(<GeneralForecast weatherToday={weatherToday} weatherTomorrow={weatherTomorrow} />)
        expect(wrapper).toMatchSnapshot()
    })

})

GeneralForecast.propTypes = {
    weatherToday: PropTypes.array.isRequired,
    weatherTomorrow: PropTypes.array.isRequired
}