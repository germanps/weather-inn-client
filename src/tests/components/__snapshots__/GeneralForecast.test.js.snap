// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in <GeneralForecast /> should show the component correctly 1`] = `
<Fragment>
  <EuiFlexGroup
    className="general-forecast"
  >
    <EuiFlexItem
      className="general-forecast__column"
    >
      <h3
        className="subtitle-low"
      >
        Previsto para hoy
      </h3>
      <p
        className="text"
      >
        today
      </p>
    </EuiFlexItem>
    <EuiFlexItem
      className="general-forecast__column"
    >
      <h3
        className="subtitle-low"
      >
        Previsión para mañana
      </h3>
      <p
        className="text"
      >
        tomorrow
      </p>
    </EuiFlexItem>
  </EuiFlexGroup>
</Fragment>
`;
